spec/models/miq_filter_spec.rb

```diff
@@ -0,0 +1,52 @@
+describe MiqFilter do
+  before do
+    @ems = FactoryGirl.create(:ems_vmware, :name => 'ems')
+    @ems_folder_path = "/belongsto/ExtManagementSystem|#{@ems.name}"
+    @datacenter = FactoryGirl.create(:ems_folder, :name => "Datacenters")
+    @datacenter.parent = @ems
+    @mtc = FactoryGirl.create(:ems_folder, :name => "MTC", :is_datacenter => true)
+    @mtc.parent = @datacenter
+    @ems_folder_path = "/belongsto/ExtManagementSystem|#{@ems.name}"
+    @mtc_folder_path = "#{@ems_folder_path}/EmsFolder|#{@datacenter.name}/EmsFolder|#{@mtc.name}"
+
+    @host_folder         = FactoryGirl.create(:ems_folder, :name => "host")
+    @host_folder.parent  = @mtc
+
+    @host_1 = FactoryGirl.create(:host_vmware, :name => "Host_1", :ext_management_system => @ems)
+    @host_2 = FactoryGirl.create(:host_vmware, :name => "Host_2", :ext_management_system => @ems)
+    @host_3 = FactoryGirl.create(:host_vmware, :name => "Host_3")
+
+    @host_1.parent = @host_folder
+    @host_2.parent = @host_folder
+    @mtc_folder_path_with_host_folder = "#{@mtc_folder_path}/EmsFolder|host"
+    @mtc_folder_path_with_host_1 = "#{@mtc_folder_path_with_host_folder}/Host|#{@host_1.name}"
+  end
```

Comment:
Can you convert a lot of these to memoized helpers instead? Makes life easier down the road when refactoring and shifting needs to occur. https://www.relishapp.com/rspec/rspec-core/v/3-4/docs/helper-methods/let-and-let